src/dos/program_serial.cpp

```diff
@@ -176,7 +176,8 @@ void SERIAL::Run()
 		}
 		if (serialports[port_index] != nullptr) {
 			serialports[port_index]->serialType = desired_type;
-			serialports[port_index]->commandLineString = commandLineString;
+			serialports[port_index]->commandLineString = std::move(
+			        commandLineString);
```

Comment:
Not sure why this was done, but since commandLineString is going out of scope anyhow, I would suggest:
```suggestion
			serialports[port_index]->commandLineString.swap(commandLineString);
```

Reply:
PVS-Studio suggested a move or swap. Does it really matter which one we use?

Reply:
TL;DR: never mind. :smile:

This basically came from some talks about abuse of std::move, but that's not the case here.

For std::string, a swap might save an allocation later on if the string is reused, but in this case I don't think there is much of a difference. commandLineString gets deleted anyhow.

Which basically negates my remark about things going out of scope in my original remark... 🤦